engine/src/main/java/org/pentaho/di/trans/steps/fileexists/FileExists.java

```diff
@@ -0,0 +1,200 @@
+
+package org.pentaho.di.trans.steps.fileexists;
+
+import org.apache.commons.vfs2.FileType;
+import org.pentaho.di.core.ResultFile;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.row.RowDataUtil;
+import org.pentaho.di.core.util.Utils;
+import org.pentaho.di.core.vfs.KettleVFS;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.trans.Trans;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.BaseStep;
+import org.pentaho.di.trans.step.StepDataInterface;
+import org.pentaho.di.trans.step.StepInterface;
+import org.pentaho.di.trans.step.StepMeta;
+import org.pentaho.di.trans.step.StepMetaInterface;
+
+/**
+ * Check if a table exists in a Database *
+ *
+ * @author Samatar
+ * @since 03-Juin-2008
+ *
+ */
+
+public class FileExists extends BaseStep implements StepInterface {
+  private static Class<?> PKG = FileExistsMeta.class; // for i18n purposes, needed by Translator2!!
+
+  private FileExistsMeta meta;
+  private FileExistsData data;
+
+  public FileExists( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr, TransMeta transMeta,
+    Trans trans ) {
+    super( stepMeta, stepDataInterface, copyNr, transMeta, trans );
+  }
+
+  public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
+    meta = (FileExistsMeta) smi;
+    data = (FileExistsData) sdi;
+
+    boolean sendToErrorRow = false;
+    String errorMessage = null;
+
+    Object[] r = getRow(); // Get row from input rowset & set row busy!
+    if ( r == null ) { // no more input to be expected...
+
+      setOutputDone();
+      return false;
+    }
+
+    boolean fileexists = false;
+    String filetype = null;
+
+    try {
+      if ( first ) {
+        first = false;
+        // get the RowMeta
+        data.previousRowMeta = getInputRowMeta().clone();
+        data.NrPrevFields = data.previousRowMeta.size();
+        data.outputRowMeta = data.previousRowMeta;
+        meta.getFields( data.outputRowMeta, getStepname(), null, null, this, repository, metaStore );
+
+        // Check is tablename field is provided
+        if ( Utils.isEmpty( meta.getDynamicFilenameField() ) ) {
+          logError( BaseMessages.getString( PKG, "FileExists.Error.FilenameFieldMissing" ) );
+          throw new KettleException( BaseMessages.getString( PKG, "FileExists.Error.FilenameFieldMissing" ) );
+        }
+
+        // cache the position of the field
+        if ( data.indexOfFileename < 0 ) {
+          data.indexOfFileename = data.previousRowMeta.indexOfValue( meta.getDynamicFilenameField() );
+          if ( data.indexOfFileename < 0 ) {
+            // The field is unreachable !
+            logError( BaseMessages.getString( PKG, "FileExists.Exception.CouldnotFindField" )
+              + "[" + meta.getDynamicFilenameField() + "]" );
+            throw new KettleException( BaseMessages.getString( PKG, "FileExists.Exception.CouldnotFindField", meta
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [KettleException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [151].  Consider adding a catch clause of this specific exception in the outer try statement.